compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/GPUTileSwizzleUtils.cpp

```diff
@@ -100,30 +100,75 @@ static void interleave(TileSwizzle &swizzle, int srcIdx, int expandedIdx) {
   swizzle.permutation = outPermutation;
 }
 
-static TileSwizzle getIntrinsicSwizzleBeforeMovingCrossThreadOutermost(
-    IREE::GPU::MMAIntrinsic intrinsic, IREE::GPU::MMAFragment fragment) {
-  auto layout = IREE::GPU::getSingleSubgroupLayout(intrinsic, fragment);
+template <typename MMAIntrinsicTy>
+static TileSwizzle getIntrinsicSwizzle(MMAIntrinsicTy intrinsic,
+                                       unsigned operandIdx) {
+  IREE::GPU::MMASingleSubgroupLayout layout;
+  const bool isScaled =
+      std::is_same<MMAIntrinsicTy, IREE::GPU::ScaledMMAIntrinsic>::value;
+  const unsigned lhsIdx = 0;
```

Comment:
Try if possible to avoid mixing signed and unsigned integers in new code. Below at line 148, unsigned rhsScalesIdx gets compared with signed operandIdx.  It's not too bad right now because it's a `==` comparison, but imagine this code being updated in the future and this becoming a `<=` for whatever reason, then this would be harder to read.

Genereally if there are preexisting integer quantities, it's OK to stick to their preexisting signedness as "don't mix different signednesses" is the overriding concern.   In independent new code, default to signed integers.